src/main/java/com/olgaboiar/guacomole/EchoClient.java

```diff
@@ -0,0 +1,45 @@
+package com.olgaboiar.guacomole;
+
+import java.io.*;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+public class EchoClient implements Closeable{
+    String host;
+    int port;
+    Socket socket;
+    int connectionAttempts;
+    int maxConnectionAttempts;
+
+    public EchoClient(String host, int port) {
+        this.host = host;
+        this.port = port;
+        this.maxConnectionAttempts = 20;
+        this.connectionAttempts = 0;
+    }
+
+    public void connect() throws IOException, InterruptedException {
+        socket = new Socket();
+        try {
+            socket.connect(new InetSocketAddress(host, port));
+        } catch (IOException e) {
+            if (this.connectionAttempts < this.maxConnectionAttempts) {
+                Thread.sleep(500);
+                this.connectionAttempts ++;
+                connect();
+            }
+        }
+    }
+
+    public byte[] send(byte[] data) throws IOException {
+        socket.getOutputStream().write(data);
+        byte[] readBuffer = new byte[data.length];
+        socket.getInputStream().read(readBuffer);
+        return readBuffer;
```

Comment:
Mildly violating single responsibility here. There are two major operations here, read and write.

Reply:
refactored in https://github.com/olgaboiar/GuacamoleServer/pull/3/commits/33a874d6b5d5f52c7d47e4675a38e82c3c72f35a